tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/ParserTestRunner.java

```
@@ -0,0 +1,115 @@
+                "send_action_source_01.bal",
+                "send_action_source_03.bal",
+                "func_def_source_08.bal",
+                "find_node_test_1.bal",
+                "enum_decl_source_08.bal",
+                "enum_decl_source_05.bal",
+                "enum_decl_source_09.bal",
+                "minutiae_test_05_with_no_newlines.bal",
+                "import_decl_source_13.bal",
+                "ambiguity_source_06.bal",
+                "ambiguity_source_11.bal",
+                "ambiguity_source_29.bal",
+                "worker_decl_source_02.bal",
+                "typed_binding_patterns_source_16.bal",
+                "forEach_stmt_source_07.bal",
+                "do_stmt_source_07.bal",
+                "match_stmt_source_13.bal",
+                "match_stmt_source_07.bal",
+                "match_stmt_source_10.bal",
+                "match_stmt_source_08.bal",
+                "match_stmt_source_12.bal",
+                "match_stmt_source_03.bal",
+                "match_stmt_source_06.bal",
+                "match_stmt_source_09.bal",
+                "match_stmt_source_11.bal"
+        );
+    }
+
+    @DataProvider(name = "parser-test-file-provider")
+    public Object[][] dataProvider() {
+        return getParserTestConfigs();
+    }
+
+    Object[][] getParserTestConfigs() {
+        List<String> skippedTests = skipList();
+        try {
+            return Files.walk(parserDir.resolve("src").resolve("test").resolve("resources"))
+                    .filter(path -> {
+                        File file = path.toFile();
+                        return file.isFile() && file.getName().endsWith(".bal")
+                                && !skippedTests.contains(file.getName());
+                    })
+                    .map(path -> new Object[]{path.toFile().getName(), path.toString()})
+                    .toArray(size -> new Object[size][2]);
+        } catch (IOException e) {
+            return new Object[0][];
```

Instead of returning an empty object, we should fail the test here.